pdl_schema.md

```diff
@@ -0,0 +1,1105 @@
+`true`.
+
+### Inlined schemas
+In pegasus, records and other named schemas need not be top-level schemas. They
+can be inlined within other record schemas.
+
+For example:
+```pdl
+namespace com.example.models
+
+import com.example.time.Date
+
+record User {
+  firstName: string
+  birthDay: optional Date
+  isActive: boolean = true
+  address: record Address {
+    state: string
+    zipcode: string
+  }
+}
+```
+`Address` record in the above example is inlined. It inherits the namespace of
+the parent record `User`, making its fully qualified name `com.example.models.Address`.
+
+* Namespace of inline types can be specified/overriden by defining a Namespace
+surrounding the inlined type.
+* The default value of fields using inlined types can be expressed using its
+serialized JSON representation.
+
+For example:
+```pdl
+namespace com.example.models
+
+import com.example.time.Date
+
+record User {
+  firstName: string
+  birthDay: optional Date
+  isActive: boolean = true
+  address: {
+    namespace com.example.models.address
+
+    record Address {
+      state: string
+      zipCode: string
+    }
+  } = {
+    "state": "CA",
+    "zipCode": "12345"
+  }
+}
+```
+***Note:*** If a record or a named schema is referenced by other schemas, it should
+be a top-level schema. Referencing in-line schemas outside the schema in which
+they are defined is not allowed.
+
+### Doc Strings
+
+Pegasus types and fields may be documented using “doc strings” following the
+Java style comments.
+* Comments using `/** */` syntax are treated as schema documentation. They will
+be included in the in-memory representation and the generated binding classes.
+* Comments using `/* */` or `//` syntax are allowed but not treated as schema
+documentation.
+
+For example:
+```pdl
+namespace com.example.models
+
+import com.example.time.Date
+
+/**
+ * A record representing an user in the system.
+ */
+record User {
+  /** First name of the user */
+  firstName: string
+
+  /** User's birth day */
+  birthDay: optional Date
+
+  // TODO: Can this be an enum?
+  /** Status of the user. */
+  isActive: boolean = true
+}
+```
+
+### Deprecation
+Pegasus supports marking types or fields as deprecated by adding `@deprecated`
+annotation. The deprecation details from the schema will be carried over to the
+bindings generated in  different languages.
+In Java language, the classes, getter and setter methods generated for
+deprecated types will be marked as deprecated.
+
+It is recommended to specify a string describing the reason for deprecation and
+the alternative as the value for the `@deprecated` annotation.
+
+Deprecate a field:
+```pdl
+namespace com.example.models
+
+import com.example.time.Date
+
+record User {
+  firstName: string
+
+  @deprecated = "Use birthDay instead."
+  birthYear: int
+
+  birthDay: Date  
+}
+```
+Deprecate a record:
+```pdl
+namespace com.example.models
+
+@deprecated = "Use Person type instead."
+record User {
+  firstName: string
+}
+```
```

Comment:
Should we provide an example of deprecation which doesn't have any string value?

Reply:
I went back and forth on this: Should this doc provide examples for all supported features?.

I finally decided to skip some features that are not recommended and have better alternative.
This was one of them. Another example is specifying default for optional field.